src/Controls/tests/TestCases.HostApp/Issues/Bugzilla/Bugzilla31366.cs

```diff
@@ -32,7 +32,7 @@ public MainPage()
 
 				popOnAppearing.Appearing += async (sender, args) =>
 				{
-					await Task.Yield();
+					await Task.Delay(50);
```

Comment:
Could we wire into an event on the page we're popping back to and add a label?

like
- push
- pop from appearing
- and then incremented a counter or inside "OnNavigatedTo" we could add a label indicating it was popped back to?

Reply:
The cause of the issue is performing the pop in the Page.Appearing event. As an alternative, I tried this approach:
![image](https://github.com/user-attachments/assets/07436a27-8ec1-4be5-862e-62790a95ac1e)
This checks the Modal stack before popping, similar to a push-then-pop pattern. Can we proceed with this approach?